modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/AbstractCSharpCodegen.java

```diff
@@ -307,10 +307,26 @@ public void processOpts() {
                 }
             }
         }
+
+        processUsings(objs);
+
         // process enum in models
         return postProcessModelsEnum(objs);
     }
 
+    private void processUsings(Map<String, Object> objs) {
```

Comment:
So in principal it sounds like the fix should be in code, to ensure the references to Model objects are always fully qualified?

If the fix was in the templates, prepending the namespace to model references, then it would still be at risk of the same issue where someone copied the templates, modified them and removed the namespace. Also the templates would potentially need to either understand the difference between Model references and core references to it didn't prefix the wrong ones.

If that makes sense :)